### PR TITLE
chore: add tsdoc for demo

### PR DESCRIPTION
**Pull request by shawnyu5**
This is a demo as to why we should "strongly encourage" proper documentation in the starchart codebase

As a developer using this library code, or code written by other people, I do not want to:

- Read through the function code to understand the logic
- Read through the tests to understand the usage of the code

I want to:

- At a glance tell what the function does
- What each of the arguments do/mean

When developing or browsing the code base, using tools like the LSP can also help in understanding the code. Example using "on hover"

image

## The solution

This is something we could "strongly enforce" (place a note in the `CONTRIBUTING.md`, and people should keep an eye on in code reviews), but not something we say all devs must follow.

Due to the "cultural" differences in each year of the open source class, enforcing excessive policies means they will eventually go unenforced. But there is a fine line between cultural differences and enforcing good coding practices for the sake of scalability and maintainability.

### CODE REVIEW

File: app/lib/notifications.server.ts [hunk 1]
```diff
@@ -43,6 +59,6 @@ const sendNotification = async (emailAddress: string, subject: string, text: str
   } catch (error) {
     logger.warn('Unable to send notification', error);
```
[the remark this answers is not in the document]
Comment (shawnyu5) [reply app/lib/notifications.server.ts:60]: Uhh I did not write this code. Pretty sure this is @SerpentBytes

[the remark this answers is not in the document]
Comment (humphd) [reply app/lib/notifications.server.ts:60]: Any code landed in the tree is "our" code, which is why @dadolhay said "we."  No need to assign blame, but when code gets back into the light (i.e., when we re-review it for other reasons), it's normal to ask questions about it.  Most code sleeps until it breaks, so it's useful to ask questions whenever something shifts in the tree.

[the remark this answers is not in the document]
Comment (shawnyu5) [reply app/lib/notifications.server.ts:60]: Im just saying the original author might be able to shed some light on why he wrote it he did it the way he did. Not saying anything is wrong with the code. I just don't understand why...

[the remark this answers is not in the document]
Comment (humphd) [reply app/lib/notifications.server.ts:60]: By that logic, it's probably useful to pull in the [original reviewers](https://github.com/DevelopingSpace/starchart/pull/125): @dadolhay and me.  Again, this is what @dadolhay is doing by asking, "why are we ignoring..."  We own all the code, and we have to maintain it.

**Comment by SerpentBytes** [comment]
@shawnyu5 I appreciate you shedding light on an error in one of my contributions to our codebase. 

I will fix this as soon as I land my active PR; however,  as one of the contributors, you are more than welcome to improve the code in question. 

We are programmers, but we are humans first; hence mistakes are inevitable.

File: app/lib/notifications.server.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dadolhay) [reply app/lib/notifications.server.ts:60]: Okay, so if we are unable to initialize the transport, I think we should make this a hard error, there should be no reason why we'd want to rescue this at this point. If code that builds on this deems the notification 'non-essential', they should be able to rescue also, but changing this as follows allows other code to potentially not handle that exception and fail (as designed)

```suggestion
    logger.error('Unable to initialize nodemailer transport', error);
    throw error;
```

**Review by dadolhay — CHANGES_REQUESTED** [review]

**Comment by dadolhay** [comment]
With the above fixed in #322 I'm OK with the change. Please resolve conflicts and I'll approve

**Comment by humphd** [comment]
It looks like this hasn't moved forward.  For the sake of getting 1.0a finished, I'm going to close this, and if you want to continue work on TS Docs stuff, we can do it in new issues.

**Pull request closed without merging**